test/unit/format_capture_helper_test.rb

```diff
@@ -0,0 +1,87 @@
+require_relative "../test_helper"
+
+module MockCaptureHelper
+  def content_for(_name, content, _options, &_block)
+    content
+  end
+
+  def capture(&block)
+    block.call
+  end
+
+  def render(*_args, &block)
+    "<g>" + block.call + "</g>"
+  end
+end
+
+class MockClass
+  include MockCaptureHelper
+end
+
+module SmartAnswer
+  class FormatCaptureHelperTest < ActiveSupport::TestCase
+    def assert_captured_content(content, format, expected)
+      options = format.present? ? { format: format } : {}
+      args = [:name, expected, options]
+
+      MockClass.any_instance.expects(:content_for).with(*args)
+
+      @test_obj.content_for(:name, options) { content }
+    end
+
+    context "class is extended by helper" do
+      setup do
+        @test_obj = MockClass.new
+        @test_obj.extend(SmartAnswer::Helpers::FormatCaptureHelper)
```

Comment:
Can we do this in the class definition?

Reply:
Had a look, turns out extend in the class definition will add the methods as Class methods than object methods. So will need call it on the instance object.

Reply:
Aha, yes, I see. We should be able to `include` it in the class definition, though. That would be more consistent with how we use helper modules outside of tests, too.

Reply:
So for this case its consistent with how we are using the Helper modules in ErbRenderer. We only want to add these methods to a particular instance of ActiveView::Base used by ErbRenderer, don't want to include them for all instances.

Reply:
OK. I see where you're coming from 👍.